### PR TITLE
CHAOS-5310 Discovered service and network maps

### PR DESCRIPTION
**Pull request by SmritiSatya**
Service discovery and network maps docs

### CODE REVIEW

**Review by amityt — DISMISSED** [review]

**Review by shovanmaity — APPROVED** [review]

File: docs/chaos-engineering/features/network-maps/intro-network-map.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
+---
+title: Introduction
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/network-maps/intro-network-map.md:2]: Might consider updating this title

File: docs/chaos-engineering/features/network-maps/intro-network-map.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+This section describes network maps, their significance, and how to create, edit, and delete them.
+
+## Before you begin
+
+- [Service discovery](/docs/chaos-engineering/features/service-discovery/intro-service-discovery.md)
+
+### What is a network map?
+
+A network map is a resource on the Harness platform (also known as the control plane) used to bind multiple interacting [discovered services](/docs/chaos-engineering/features/service-discovery/intro-service-discovery.md) into a single object, thereby allowing the users to act (test, monitor, deploy, etc.) on a group of services.
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/network-maps/intro-network-map.md:15]:
```suggestion
A network map is a resource on the Harness Platform (also known as the control plane) used to bind multiple interacting [discovered services](/docs/chaos-engineering/features/service-discovery/intro-service-discovery.md) into a single object, thereby allowing the users to act (test, monitor, deploy, and so on) on a group of services.
```

File: docs/chaos-engineering/features/network-maps/intro-network-map.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+### Why is a network map required?
+
+Network maps help:
+
+- Test, monitor, and deploy a group of services.
+- Determine the resilience coverage and resilience score for the overall network map and its constituent services
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/network-maps/intro-network-map.md:22]:
```suggestion
- Determine the resilience coverage and resilience score for the overall network map and its constituent services.
```

File: docs/chaos-engineering/features/network-maps/intro-network-map.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    ![](./static/service-rel-4.png)
+
+5. Select **Save**.
+
+    ![](./static/save-nw-5.png)
+
+### Edit network map
+
+To edit a network map, select the vertical menu and select **Edit**. Make the necessary changes and select **Save**.
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/network-maps/intro-network-map.md:56]:
```suggestion
To edit a network map, select **More Options** (&vellip;), then select **Edit**. Make the necessary changes and select **Save**.
```

File: docs/chaos-engineering/features/network-maps/intro-network-map.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+![](./static/edit-8.png)
+
+### Delete network map
+
+1. To delete a network map, select the vertical menu and select **Delete**.
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/network-maps/intro-network-map.md:62]:
```suggestion
1. To delete a network map, select **More Options** (&vellip;), then select **Delete**.
```

File: docs/chaos-engineering/features/service-discovery/intro-service-discovery.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
+---
+title: Introduction
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/service-discovery/intro-service-discovery.md:2]: might consider updating this title

File: docs/chaos-engineering/features/service-discovery/intro-service-discovery.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+sidebar_position: 1
+description: An introduction to service discovery and its significance
+---
+
+This section introduces you to **service discovery** for **Kubernetes infrastructure**, its significance, and its purpose.
+
+### What is a discovered service?
+Discovered service is an entity on the Harness platform (also known as the control plane) that corresponds to a Kubernetes service on your (user) cluster. It includes details about the connections made from and to it.
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/service-discovery/intro-service-discovery.md:10]:
```suggestion
Discovered service is an entity on the Harness Platform (also known as the control plane) that corresponds to a Kubernetes service on your (user) cluster. It includes details about the connections made from and to it.
```

File: docs/chaos-engineering/features/service-discovery/intro-service-discovery.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+1. Scan your Kubernetes cluster periodically (you can define the interval or ad-hoc).
+2. Build a database of services that describes the relationship between the services.
+3. Provide APIs to group the discovered services into a map, wherein the map represents a topological view of an application.
+
+### How does HCE leverage discovered services?
+
+HCE tests the resilience of the application, and HCE uses the discovered services to:
+
+- Identify the various available services in the chaos module, that is, the chaos targets in the Kubernetes cluster that you can deploy using Harness or other means.
+- Record the resources backing the above services (logical resources such as deployment, pods, containers, processes, FQDNs, ports, physical resources such as nodes, storage, etc.)
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/service-discovery/intro-service-discovery.md:32]:
```suggestion
- Record the resources backing the above services (logical resources such as deployment, pods, containers, processes, FQDNs, ports, physical resources such as nodes, storage, and so on).
```

File: docs/chaos-engineering/features/service-discovery/service-discovery-usage.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
+---
+title: Customize discovery agent
+sidebar_position: 2
+description: Customize discovery agent
+---
+
+This section describes how you can [create](#customize-discovery-agent), [edit](#edit-discovery-agent), and [delete](#delete-discovery-agent) a discovery agent.
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/service-discovery/service-discovery-usage.md:7]:
```suggestion
This topic describes how you can [create](#customize-discovery-agent), [edit](#edit-discovery-agent), and [delete](#delete-discovery-agent) a discovery agent.
```

File: docs/chaos-engineering/features/service-discovery/service-discovery-usage.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+
+## Before you begin, review the following
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/service-discovery/service-discovery-usage.md:9]:
```suggestion
### Before you begin
```

File: docs/chaos-engineering/features/service-discovery/service-discovery-usage.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+- [Introduction to service discovery](/docs/chaos-engineering/features/service-discovery/intro-service-discovery.md)
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/service-discovery/service-discovery-usage.md:11]:
```suggestion
Review the [Introduction to service discovery](/docs/chaos-engineering/features/service-discovery/intro-service-discovery.md).
```

File: docs/chaos-engineering/features/service-discovery/service-discovery-usage.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+When you are onboarding, one of the steps involves discovering services. HCE creates the discovery agent that automatically discovers services for your application.
+
+If you want to customize the discovery agent, follow the steps below.
+
+### Customize Discovery Agent
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/service-discovery/service-discovery-usage.md:17]:
```suggestion
### Customize discovery agent
```

File: docs/chaos-engineering/features/service-discovery/service-discovery-usage.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+1. To customize the discovery agent, navigate to **Chaos** and select **Discovery**. Click **New Discovery Agent**.
+
+    ![](./static/usage/discovery.png)
+
+2. Select an **environment**, **infrastructure**, **Discovery Agent Name** and **namespace**. The next step (optional) requires information such as node selector, blacklisted namespaces, and period of detecting the services. Select **Create New Discovery Agent**.
+
+    ![](./static/usage/add-details-discovery.png)
+
+### Edit Discovery Agent
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/service-discovery/service-discovery-usage.md:27]:
```suggestion
### Edit discovery agent
```

File: docs/chaos-engineering/features/service-discovery/service-discovery-usage.md [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+
+1. To edit a discovery agent, navigate to the agent and click **Edit**. Make the necessary changes to the required fields.
+
+    ![](./static/usage/edit-dis-agent-1.png)
+
+2. Select **Update Discovery Agent**.
+
+    ![](./static/usage/edit-details-discovery.png)
+
+### Delete Discovery Agent
```
Comment (brian-f-harness) [inline docs/chaos-engineering/features/service-discovery/service-discovery-usage.md:37]:
```suggestion
### Delete discovery agent
```

**Review by brian-f-harness — APPROVED** [review]
A few optional suggestions, otherwise looks good

**Pull request merged**